### PR TITLE
Feature/int 692 add ability to pass user id from web app

### PR DESCRIPTION
**Pull request by DmytroHryshyn**

### CODE REVIEW

File: src/components/user/userAccountStorage.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import * as vscode from 'vscode';
+
+export class GlobalStateAccountStorage {
+	constructor(private readonly globalState: vscode.Memento) {}
```
Comment (grzpab) [inline src/components/user/userAccountStorage.ts:4]: I usually write it like `__globalState`

File: src/components/user/userService.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	}
+}
+
+export interface UserAccountStorage {
+	getUserAccount(): string | null;
+	setUserAccount(value: string | undefined): void;
+}
+
+export class UserService {
+	constructor(private readonly storage: UserAccountStorage) {}
```
Comment (grzpab) [inline src/components/user/userService.ts:13]: `__storage`

File: src/extension.ts [hunk 3]
```diff
@@ -928,27 +935,34 @@ export async function activate(context: vscode.ExtensionContext) {
...
+					vscode.window.showTextDocument(document);
+				}
 
-				vscode.window.showTextDocument(document);
+				if (userId) {
+					try {
+						userService.linkUsersIntuitaAccount(userId);
+					} catch (e) {
+						if (e instanceof AlreadyLinkedError) {
+							// @TODO show warning asking to replace with new account
```
Comment (grzpab) [inline src/extension.ts:962]: I think you can use window.showNotification or similar here.

File: src/extension.ts [hunk 4]
```diff
@@ -928,27 +935,34 @@ export async function activate(context: vscode.ExtensionContext) {
 			handleUri: async (uri) => {
 				const searchParams = new URLSearchParams(uri.query);
 				const base64UrlEncodedContent = searchParams.get('c');
+				const userId = searchParams.get('userId');
```
Comment (grzpab) [inline src/extension.ts:938]: does it need to be base64urlencoded?

File: src/components/user/userAccountStorage.ts (hunk 1 above)
Comment (DmytroHryshyn) [reply src/components/user/userAccountStorage.ts:4]: so, for all private vars we should use `__` prefix, right?

File: src/extension.ts (hunk 4 above)
Comment (DmytroHryshyn) [reply src/extension.ts:938]: i think no

**Review by grzpab — APPROVED** [review]

**Pull request merged**